core/client/src/main/java/alluxio/client/block/stream/LocalFilePacketReader.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.block.stream;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+import alluxio.worker.block.io.LocalFileBlockReader;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+
+import java.io.IOException;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * A packet reader that simply reads packets from a local file.
+ */
+@NotThreadSafe
+public final class LocalFilePacketReader implements PacketReader {
```

Comment:
yeah, LocalFileBlockReader can be deprecated if we deprecate the legacy implementations the data transfer protocol. For now, I cannot since i don't want to change the behavior of legacy impl.